### PR TITLE
Discuss selection of client identifiers

### LINKED ISSUE #65: email by Martin Thomson: contextual identity
Martin also sent some additional feedback by email:

> This document feels incomplete.  I'll give a few examples of applications of the general concept that aren't covered.
> 
> Separating identity from the right to communicate is a major concept that is not as fully explored as it could be.
> 
> This is something that the draft does discuss in a way, because it talks about using intermediaries to shield the identity of entities that wish to initiate communication.
> 
> The entire class of use for pseudonymous identifiers isn't really given a solid treatment.  I work with people who provide email aliases for privacy purposes.  There, the underlying concept is that an email address typically serves two distinct functions: a) an identifier, and b) authorization to communicate.
> 
> When creating temporary email addresses, especially when these are unique for each context, we break the identification part (a).  We give the recipient the means to communicate with us, but deny them the means to identify.  Identification means correlation across multiple contexts.  (As an aside, that's a failing in RFC 6973, which treats identification and correlation as distinct threats rather than facets of the same and distinct only in degree.)
> 
> A contextually-bound pseudonymous identifier provides no correlating power. Again, this relies on having intermediaries that can be trusted not to collude and reveal the true identity, so I believe that this qualifies as a partitioning.
> 
> The same basic idea also underpins Web Push (RFC 8030).  Websites are given the means to send push messages to user agents, but are denied the means of identification.  Here, the privacy guarantees are stronger than email because the intermediary does not need to be entrusted with viewing the message contents.  End-to-end protected email is possible, but email comes with very different expectations.
> 
> I believe that this identity separation concept is also a property that is inherent to ICN architectures, but I'm not as qualified to comment on that as others, including some IAB members.  Similarly, I believe that it would be possible to more fully capitalize on randomized IPv6 ULAs (RFC 4941), but there are folks on the IAB better positioned to handle that as well.
> 
> Finally, it is maybe not in scope, but I wonder how the authors perceive the relation between this work and contextual integrity.  It seems like maybe this is a foundational principle that is implied throughout, perhaps along with other principles.  However, it's not really clear which aspects of privacy are potentially responsive to the types of partitioning that are discussed.  There is missed opportunity here, I think. Just looking at specific privacy risks (those listed in Section 5 of RFC 6973 are a good start), it is obvious how partitioning helps (or doesn't) with some, but not all.

### PR DESCRIPTION
**Pull request by tfpauly**
Closes #65

### CODE REVIEW

File: draft-iab-privacy-partitioning.md [hunk 1]
```diff
@@ -607,6 +607,27 @@ Depending on the application and system constraints, users may not be able to pr
 in privacy contexts. As a result, fingerprinting information, when combined with non-user-identifying
 user data, could promote user data to user-identifying information.
 
+## Selecting Client Identifiers
+
+The selection of client identifiers used in the contexts used for privacy partitioning has a large
+impact on the effectiveness of partitioning. Identifier selection can either undermine or improve
+the value of partitioning.
```
Comment (mirjak) [inline draft-iab-privacy-partitioning.md:614]:
```suggestion
the value of partitioning. Usually each context uses some kind of identifier which may are may not be directly
associated with a client identity, e.g. a pseudonym or random one-time identifier could be used as well.
```

**Review by mirjak — CHANGES_REQUESTED** [review]
I proposed one addition to clarify that each context has some kind of identifier. But otherwise this looks fine.

File: draft-iab-privacy-partitioning.md (hunk 1 above)
Comment (tfpauly) [reply draft-iab-privacy-partitioning.md:614]:
```suggestion
the value of partitioning. Generally, each context involves some form of client identifier,
which might be directly associated with a client identity, but can also be a pseudonym
or a random one-time identifier.
```

File: draft-iab-privacy-partitioning.md [hunk 2]
```diff
@@ -607,6 +607,29 @@ Depending on the application and system constraints, users may not be able to pr
...
+which might be directly associated with a client identity, but can also be a pseudonym
+or a random one-time identifier.
+
+Using the same client identifier across multiple contexts can partly or wholly undermine the
+effectiveness of partitioning, by allowing the various contexts to be linked back to the same client.
+For example, if a client uses proxies as described in {{masque}} to separate connections, but uses
+the same email address to authenticate to two servers in different contexts, those actions can be linked
+back to the same client. While this does not undermine all of the partitioning achieved through
+proxying (the contexts along the network path still cannot correlate the client identity and
+what servers are being accessed), the overall effect of unlinkability is diminished.
```
Comment (chris-wood) [inline draft-iab-privacy-partitioning.md:624]:
```suggestion
what servers are being accessed), the overall effect of partitioning is diminished.
```

**Review by chris-wood — APPROVED** [review]
Lovely!

**Pull request merged**